### ISSUE TITLE
Loading with OnlyNeeded option

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/agt#11]
I looked at the issue mentioned in https://github.com/gap-system/gap-distribution/pull/104.

What happens is an interaction between Grape and Digraphs, caused by the fact that when we call `LoadPackage("agt" : OnlyNeeded)`, then `OnlyNeeded` becomes a global option. Now agt dependencies have 

    NeededOtherPackages := [["GAPDoc", "1.5"],["grape","4.8"],["design","1.6"],["Digraphs","0.12.2"]],

and each of those packages will be loaded with the same `OnlyNeeded` option (I suppose). But Digraphs has this:

```
BindGlobal("DIGRAPHS_IsGrapeLoaded",
           IsPackageMarkedForLoading("grape", "4.8.1"));

if not DIGRAPHS_IsGrapeLoaded then
  IsGraph := ReturnFalse;
  Vertices := IdFunc;
  Adjacency := IdFunc;
fi;
```
and apparently `DIGRAPHS_IsGrapeLoaded=false` and `IsGraph` and `Vertices` are then defined as above. Next, GAP tries to load GRAPE as needed package and clearly this causes an error.

Any ideas how to fix this @fingolfin @james-d-mitchell ? One is to withdraw tests with `OnlyNeeded` option (or let them fail without failing the test) - but they are useful to catch undocumented dependencies. Another is to review relations between GRAPE and Digraphs - the change to allow Digraphs to load without GRAPE has been made 5 years ago, do we still need it?

**Comment by james-d-mitchell** [comment]
@alex-konovalov I didn't see this until now, I'll have to give some thought to this.

**Comment by olexandr-konovalov** [comment]
@james-d-mitchell thanks, that would be very useful!

**Comment by fingolfin** [comment]
This has been fully fixed in digraphs 1.5.2
